Develop2

shemolesa opened the pull request:
Созданы тесты на эндпоинты сервиса: создание пользователя, логин пользователя, изменение данных пользователя, создание заказа, получение заказов пользователя

Commit:
Созданы тесты на эндпоинты сервиса: создание пользователя, логин пользователя, изменение данных пользователя,
создание заказа, получение заказов пользователя

shemolesa committed:
Update README.md

исправлена опечатка

AlexKitIt commented on conftest.py:
```diff
@@ -0,0 +1,49 @@
+
+@allure.step('Регистрация нового тестового покупателя, авторизация и последующее удаление')
+@pytest.fixture()
+def response_customer():
+    # отправляем запрос на регистрацию тестового покупателя и сохраняем ответ в переменную
+    payload = generate_data_customer()
+    # отправляем запрос на регистрацию тестового покупателя и сохраняем ответ в переменную
+    response_customer = requests.post(URL_CUSTOMER_REG, data=payload)
+    # отправляем запрос на авторизацию тестового покупателя и сохраняем ответ в переменную
```
Нужно исправить: большая часть этого метода уже описана в фикстуре ниже, необходимо передавать ее на вход и дополнять а не копипастить

AlexKitIt commented on conftest.py:
```diff
@@ -0,0 +1,49 @@
+    response_delete = requests.delete(URL_CUSTOMER, headers={'Authorization': access_token})
+
+
+# @allure.step('Регистрация нового тестового покупателя с передачей его id и с последующим удалением')
+# @pytest.fixture()
+# def response_courier():
+#     # отправляем запрос на регистрацию тестового курьера и сохраняем ответ в переменную
+#     response_courier = requests.post(data.URL_COURIER, data.TEST_COURIER)
+#     # отправляем запрос на авторизацию тестового курьера и сохраняем ответ в переменную
+#     response_id_courier = requests.post(data.URL_COURIER_LOGIN, data.TEST_COURIER)
+#     # находим id курьера
+#     id_courier = response_id_courier.json()['id']
+#     yield response_courier, response_id_courier
+#     #удаляем тестового курьера
+#     response_delete = requests.delete(f'{data.URL_COURIER}/{id_courier}')
```
Нужно исправить: закоммеченного кода быть не должно

AlexKitIt commented on helpers.py:
```diff
@@ -0,0 +1,53 @@
+    random_ingredients = random.choice(list_ingredients)
+    return random_ingredients
+
+@allure.step('Регистрация курьера')
+def customer_registration():
+    payload = generate_data_customer()
+    response_customer_registration = requests.post(URL_CUSTOMER_REG, data=payload)
+    return response_customer_registration, payload
+
+@allure.step('Авторизация курьера')
+def customer_authorization(new_customer, payload):
+    access_token = new_customer.json()['accessToken']
+    response_customer_authorization = requests.post(URL_CUSTOMER_LOGIN, data=payload,
+                                           headers={'Authorization': access_token})
+    return response_customer_authorization, payload
+
+
+@allure.step('Удаление курьера')
+def delete_customer(registered_customer, payload):
+    access_token = registered_customer[0].json()['accessToken']
+    response_delete = requests.delete(URL_CUSTOMER, data=payload,
+                                           headers={'Authorization': access_token})
+    return response_delete
```
Нужно исправить: логические методы в helpers не находятся, под них пишется отдельный модуль

AlexKitIt commented on tests/test_customer.py:
```diff
@@ -0,0 +1,85 @@
+        response_authorization = requests.post(URL_CUSTOMER_LOGIN,
+                                               headers={'Authorization': response_fixture[1]}, data=payload)
+        # проверяем, что замена выполнилась успешно и пользователь с новым паролем может авторизоваться
+        assert response_changing.status_code == 200 and response_authorization.status_code == 200
```
Нужно исправить: здесь и далее: одной проверки на статус код недостаточно, необходимо парсить ответ

AlexKitIt commented on tests/test_customer.py:
```diff
@@ -0,0 +1,85 @@
+        # меняем пароль у тестового покупателя
+        response_changing = requests.patch(URL_CUSTOMER, headers={'Authorization': response_fixture[1]}, json=new_email)
+        # удаляем нового покупателя
+        response_delete = requests.delete(URL_CUSTOMER,
```
Нужно исправить: здесь и далее: переменная создается но нигде не используется

Commit:
Переписана фикстура response_customer: на вход подается фикстура response_customer_unauthorized
Удален закомменченный код
Удалены неиспользуемые методы helpers
Добавлена проверка тела ответа в тестах test_changing_customer_data_successfully и test_changing_customer_password_successfully
Удалено создание неиспользуемой переменной response_delete

Commit:
Переписана фикстура response_customer: на вход подается фикстура response_customer_unauthorized
Удален закомменченный код
Удалены неиспользуемые методы helpers
Добавлена проверка тела ответа в тестах test_changing_customer_data_successfully и test_changing_customer_password_successfully
Удалено создание неиспользуемой переменной response_delete

Commit:
Merge

AlexKitIt commented on tests/test_customer.py:
```diff
@@ -0,0 +1,94 @@
+        response_authorization = requests.post(URL_CUSTOMER_LOGIN,
+                                               headers={'Authorization': response_fixture[1]}, data=payload)
+        # проверяем, что замена выполнилась успешно и пользователь с новым паролем может авторизоваться
+        assert ((response_changing.status_code == 200 and response_changing.json()['user']['name'] == payload['name']
```
Столько условий писать в одном assert не стоит, не понятно будет на чем именно упал тест, не больше пары условий на один assert, нужно больше проверок - пишем еще один
